commandsv3/src/test/java/org/wpilib/commands3/SchedulerTimingTests.java

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package org.wpilib.commands3;
+
+import static edu.wpi.first.units.Units.Microseconds;
+import static edu.wpi.first.units.Units.Milliseconds;
+import static edu.wpi.first.units.Units.Seconds;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import edu.wpi.first.wpilibj.RobotController;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.Test;
+
+class SchedulerTimingTests extends CommandTestBase {
+  @Test
+  void commandAwaitingItself() {
+    // This command deadlocks on itself. It's calling yield() in an infinite loop, which is
+    // equivalent to calling Coroutine.park(). No deleterious side effects other than stalling
+    // the command
+    AtomicReference<Command> commandRef = new AtomicReference<>();
+    var command =
+        Command.noRequirements().executing(co -> co.await(commandRef.get())).named("Self Await");
+    commandRef.set(command);
+
+    m_scheduler.schedule(command);
+    m_scheduler.run();
+    assertTrue(m_scheduler.isRunning(command));
+  }
+
+  @Test
+  void commandDeadlock() {
+    AtomicReference<Command> ref1 = new AtomicReference<>();
+    AtomicReference<Command> ref2 = new AtomicReference<>();
+
+    // Deadlock scenario:
+    // command1 starts, schedules command2, then waits for command2 to exit
+    // command2 starts, waits for command1 to exit
+    //
+    // Each successive run sees command1 mount, check for command2, then yield.
+    // Then sees command2 mount, check for command1, then also yield.
+    // This is like two threads spinwaiting for the other to exit.
+    //
+    // Externally canceling command2 allows command1 to continue
+    // Externally canceling command1 cancels both
```

Comment:
Since these commands have a parent-child relationship, that should be reflected in their (variable) naming.

Also, a similar test without the parent-child relationship (ie, schedule both independently and then have then await one on another) would be good.